source/documentation/other-topics/troubleshooting.md

```diff
@@ -0,0 +1,111 @@
+
+##### Cause
+A `CrashloopBackOff` means that you have a pod starting, crashing, starting again, and then crashing again.
+
+A PodSpec has a restartPolicy field with possible values Always, OnFailure, and Never which applies to all containers in a pod. The default value is Always and the restartPolicy only refers to restarts of the containers by the kubelet on the same node (so the restart count will reset if the pod is rescheduled in a different node). Failed containers that are restarted by the kubelet are restarted with an exponential back-off delay (10s, 20s, 40s …) capped at five minutes, and is reset after ten minutes of successful execution. This is an example of a PodSpec with the restartPolicy field:
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: dummy-pod
+spec:
+  containers:
+    - name: dummy-pod
+      image: ubuntu
+  restartPolicy: Always
+```
+
+The main causes of `Crashloopback` are:
+
+- The application inside the container keeps crashing.
+- Some type of parameters of the pod or container have been configured incorrectly.
+- An error has been made when deploying Kubernetes.
+
+##### Troubleshooting
+As this issue is quite vague, you'll want to start by checking STDOUT using the command `kubectl -n <namespace> logs <pod_name>`. This should give you a clear understanding of whether the application has crashed. If this is the case please review the application code and the solution will become evident.
+
+##### Solution
+Fix application or misconfiguration errors.
+
+#### My pod shows `ImagePullBackOff` after deployment
+
+##### Scenario
+You have deployed an application to the Cloud Platform and its status shows `ImagePullBackOff` (you can get the status by running `kubect get pods -n <namespace>`).
+
+##### Cause
+This error is caused by a misconfiguration in your deployment, usually an invalid image `tag:`.
+
+##### Troubleshooting
+Again, utilising the `kubectl -n <namespace> describe <pod_name>` command you can see that the pod has failed to pull down the correct image:
+```
+  Normal   Pulling    10m (x4 over 12m)    kubelet, worker-node  pulling image "redis:foobar"
+  Warning  Failed     10m (x4 over 12m)    kubelet, worker-node  Error: ErrImagePull
+```
+
+##### Solution
+This can be corrected by amending the `image` in your deployment.
+
+#### I get the error 'container is unhealthy, it will be killed and re-created'
+
+##### Situation
+Your pod appears to be jumping between an `error` and `ready` state. You describe the pod and get an error `container is unhealthy, it will be killed and re-created` 
+
+##### Cause
+Kubernetes provides two essential features called Liveness Probes and Readiness Probes. Essentially, Liveness/Readiness Probes will periodically perform an action (e.g. make an HTTP request, open a tcp connection, or run a command in your container) to confirm that your application is working as intended.
```

Comment:
It would be useful to add links to the kubernetes documentation on liveness and readiness probes.

Reply:
Thanks for the review @digitalronin - I've made the requested changes.